pkg/controller/lifecycle/add.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2022 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lifecycle
+
+import (
+	"context"
+	"time"
+
+	// Local
+	"github.com/go-logr/logr"
+	"github.com/intel/gardener-extension-cri-resmgr/pkg/consts"
+	"github.com/intel/gardener-extension-cri-resmgr/pkg/options"
+
+	// Gardener
+	"github.com/gardener/gardener/extensions/pkg/controller/extension"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/controllerutils/mapper"
+
+	// Other
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// configMapToAllExtensionsMapper maps creates reconciliation requests for extensions based on dedicate configMap of cri-resmgr extension.
+func configMapToAllExtensionMapper(ctx context.Context, log logr.Logger, reader client.Reader, obj client.Object) []reconcile.Request {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	configMap, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		return nil
+	}
+
+	extensionList := &extensionsv1alpha1.ExtensionList{}
+	if err := reader.List(ctx, extensionList); err != nil {
+		return nil
```

Comment:
`configMapToAllExtensionMapper` doesn't allow to return errors - but if everything is all right it is supposed to return slice of reconcillation Requests - in case of error we return nil slice that will not start reconcillation

but you right, that such errors shouldn't go silently - I added log with warning